docs/architecture/adr-003-subaccounts.md

```diff
@@ -0,0 +1,88 @@
+# ADR 3: Module SubAccounts
+
+## Changelog
+
+## Context
+
+Currently `ModuleAccount`s must be declared upon supply keeper initialization. In addition to this they don't allow for separation of fungible coins within an account.
+
+New structs should be added so a `ModuleAccount` can dynamically add accounts.
+
+## Decision
+
+Add the following structs into `x/supply`.
+
+### Implementation Changes
+
+Introduce two new structs:
+
+* `ModuleMultiAccount`
+* `SubAccount`
+
+`ModuleMultiAccount` maintains an array of `SubAccount` as well as a list of permissions defined upon initialization of supply keeper (permAddrs).
+It has no pubkey and no limit on the number of `SubAccount`s that can be created.
+Its constructor returns a `ModuleMultiAccount` with no `SubAccount`s.
+`MultiModuleAccount` will implement the Account interface.
+
+```go
+// SetCoins will return an error to prevent ModuleMultiAccount address from having a balance.
+// SubAccounts can only be appended to the SubAccount array.
+// Passively tracks the sum of all SubAccount balances.
+type ModuleMultiAccount struct {
+    Subaccs []SubAccount
+    Permissions []string
+    Coins sdk.Coins // passively track all sub account balances
+
+    CreateSubAccount(pubkey, address) int // returns id of subaccount
+    GetSubAccount(id int) SubAccount
+}
+```
+
+To invalidate a `SubAccount` the `ModuleMultiAccount` calls `SetAccountDisabled` for a `SubAccount`
```

Comment:
TODO: Explain what is meant by "invalidating" a `SubAccount`

Reply:
deleted notion of invalidation/deletion